precompiles/precompile.go

```diff
@@ -725,7 +725,14 @@ func (p *Precompile) Call(
 			}
 			return solErr.data, callerCtx.gasLeft, vm.ErrExecutionReverted
 		}
-		return nil, callerCtx.gasLeft, errRet
+		log.Debug("precompile reverted with non-solidity error", "precompile", precompileAddress, "input", input, "err", errRet)
```

Comment:
This trips on `vm.ErrOutOfGas` and friends

Reply:
I've exempted `vm.ErrOutOfGas`. Were there any others you think should be exempted?

Reply:
I think the others are `Debug` worthy. Something like `vm.ErrWriteProtection` is noteworthy